### PR TITLE
Fix/hover styles

### LINKED ISSUE #965: Hover styling
The affordance of many interactive elements could be improved by changes in style when hovered.

E.g. 
image
image

Our general-purpose links change color when hovered, but the difference is so small it's hard to make out.

### PR DESCRIPTION
**Pull request by clari182**
- closes #965 

Fixing hover styles in:
1. Cite page
- timeline
- taxonomy edit -> Button updated to match other buttons
- Pagination buttons at bottom -> style fixed to match other buttons
2. Discover page
- List - card title
- Detail - card title

### CODE REVIEW

**Review by lmcnulty — APPROVED** [review]
This is all good, but per that issue I think all/most links on the site should change style when hovered, which they currently don't.

**Comment by clari182** [comment]
@lmcnulty I added a hover underline on links that appear inside texts. Ready for another round of review

**Comment by cesarvarela** [comment]
@clari182 conflicts time

**Comment by clari182** [comment]
conflicts fixed @cesarvarela

**Comment by cesarvarela** [comment]
@clari182 again!

File: site/gatsby-site/src/global.css [hunk 1]
```diff
@@ -78,7 +78,7 @@ a {
   text-decoration: none;
 }
 a:hover {
-  text-decoration: none;
+  text-decoration: underline;
```
Comment (lmcnulty) [inline site/gatsby-site/src/global.css:81]: Thanks! Can you also add `text-underline-offset: 4px` so it has more space to breath? After that I'm good to merge.

**Review by lmcnulty — APPROVED** [review]

**Pull request merged**